artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTStateManager.java

```diff
@@ -172,25 +175,13 @@ public String toString() {
       return "MQTTSessionStateManager@" + Integer.toHexString(System.identityHashCode(this));
    }
 
-   public void storeSessionState(MQTTSessionState state) throws Exception {
-      logger.debug("Adding durable MQTT state record for: {}", state.getClientId());
-
-      /*
-       * It is imperative to ensure the routed message is actually *all the way* on the queue before proceeding
-       * otherwise there can be a race with removing it.
-       */
-      CountDownLatch latch = new CountDownLatch(1);
-      Transaction tx = new TransactionImpl(server.getStorageManager());
-      server.getPostOffice().route(serializeState(state, server.getStorageManager().generateID()), tx, false);
-      tx.addOperation(new TransactionOperationAbstract() {
-         @Override
-         public void afterCommit(Transaction tx) {
-            latch.countDown();
-         }
-      });
-      tx.commit();
-      if (!latch.await(timeout, TimeUnit.MILLISECONDS)) {
-         throw MQTTBundle.BUNDLE.unableToStoreMqttState(timeout);
+   public void storeDurableSubscriptionState(MQTTSessionState state) throws Exception {
+      if (subscriptionPersistenceEnabled) {
```

Comment:
I would do this through configuration.isSubscriptionPersistenceEnabled() directly...


doing that way you would be able to repopulate the variable while the broker is running, and this value would take place without require a reboot.

Reply:
Well the point is, if it isn't being done now consciously and therefore not tested, then it probably should be

Reply:
Many, many configuration parameters are read when the broker starts and changing them at runtime makes no difference to the running broker. I don't see why this one should be any different.